internal/api/runme/runner/v1/runner.proto

```diff
@@ -0,0 +1,122 @@
+syntax = "proto3";
+
+package runme.runner.v1;
+
+import "google/protobuf/wrappers.proto";
+import "google/protobuf/duration.proto";
+
+option go_package = "github.com/stateful/runme/internal/gen/proto/go/runme/runner/v1;runnerv1";
+
+message Session {
+    string id = 1;
+
+    // envs keeps track of session environment variables.
+    // They can be modified by executing programs which
+    // alter them through "export" and "unset" commands.
+    repeated string envs = 2;
+
+    // metadata is a map of client specific metadata.
+    map<string, string> metadata = 3;
+}
+
+message CreateSessionRequest {
+    // metadata is a map of client specific metadata.
+    map<string, string> metadata = 1;
+
+    // envs field provides an initial set of environment variables
+    // for a newly created session.
+    repeated string envs = 2;
+}
+
+message CreateSessionResponse {
+    Session session = 1;
+}
+
+message GetSessionRequest {
+    string id = 1;
+}
+
+message GetSessionResponse {
+    Session session = 1;
+}
+
+message ListSessionsRequest {}
```

Comment:
Is there any reason we need this command? That is, is there any circumstance in which a process would want to access a session it did not create?

Imagine an attacker gets access to the service. They could then use this command to find existing sessions and print out their environment variables.

Reply:
We were thinking about interoperability between the extension and CLI where that could be helpful. Overall, having full CRUD is helpful to run experiments.

Security is a factor, but hiding this method is "security by obscurity" which is not recommended. We might hide it before figuring out something better, though.